soda_machine.rb

```diff
@@ -7,12 +7,26 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.length
   end
 
   def find_soda(soda_brand)
+    soda_check = @sodas.each do |soda|
+      if soda.brand == soda_brand
+        return soda
+      end
+    end
+
   end
 
   def sell(soda_brand)
+    if @sodas.each do |soda|
+      if soda.brand == soda_brand
+        @cash += soda.price
+        return soda
+      end
+    end
   end
```

Comment:
looks like you forgot to delete the `soda_brand` from the list of `@sodas`.  also, why not reuse the method `find_soda` from above?